sklearn/feature_selection/tests/test_mutual_info.py

```diff
@@ -207,3 +207,26 @@ def test_mutual_info_options(global_dtype):
         assert_allclose(mi_5, mi_6)
 
         assert not np.allclose(mi_1, mi_3)
+
+
+def test_mutual_information_symmetry_classif_regression():
+    """Check that `mutual_info_classif` and `mutual_info_regression` are
+    symmetric by switching the target `y` as `feature` in `X` and vice
+    versa.
+
+    Non-regression test for:
+    https://github.com/scikit-learn/scikit-learn/issues/23720
+    """
+    rng = np.random.RandomState(0)
```

Comment:
Should this use the global seed fixture (#22827)?

Reply:
I am wondering if this is not overkill here.

Reply:
¯\_(ツ)_/¯

Part of my question is to figure out what other people's feeling is for when to use the global seed thing and when not :)

Reply:
IMO, this is to test some numerical stability issues triggered in corner cases.
Here, I would not have necessarily done it because the two input vectors used to compute the MI will be used exactly with the same functions underneath.